hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsContext.java

```diff
@@ -69,4 +129,58 @@ public void registerActionListener(ActionListener listener) {
     public List<ActionListener> getActionsListeners() {
         return actionsListeners;
     }
+
+    public void notifyListeners(Set<DefinitionsEvent> notifications) {
+        Set<DefinitionsEvent.Type> notificationTypes = notifications.stream()
+                .map(n -> n.getType())
+                .collect(Collectors.toSet());
+        log.debugf("Notifying applicable listeners %s of events %s", definitionListeners, notifications);
+        definitionListeners.entrySet().stream()
+                .filter(e -> shouldNotify(e.getValue(), notificationTypes))
+                .forEach(e -> {
+                    log.debugf("Notified Listener %s of %s", e.getKey(), notificationTypes);
+                    e.getKey().onChange(notifications.stream()
+                            .filter(de -> e.getValue().contains(de.getType()))
+                            .collect(Collectors.toSet()));
+                });
+        if (!distributed) {
+            distributedListener.stream().forEach(listener -> listener.onChange(mapDistributedEvents(notifications)));
+        }
```

Comment:
If I understand, this treats a distributedListener like a definitionsListener if we are not in distributed mode?  If so, that is good.